os/fs/smartfs/README.txt

```diff
@@ -1,90 +1,179 @@
+  "sectors". Sectors have both physical and logical number assignments.
+  The physical sector number represents the actual offset from the beginning
+  of the flash partition, while the logical sector number is assigned as needed.
+  A physical sector can have any logical sector assignment and vice-versa, and
+  as files are created, modified and destroyed, the logical sector number
+  assignment for a given physical sector will change over time. The logical sector
   number is saved in the physical sector header as the first 2 bytes, and
   the MTD layer maintains an in-memory map of the logical to physical mapping.
   Only physical sectors that are in use will have a logical assignment.
 
   Also contained in the sector header is a flags byte and a sequence number.
   When a sector is allocated, the COMMITED flag will be "set" (changed from
-  erase state to non-erase state) to indicate the sector data is valid.  When
-  a sector's data needs to be deleted, the RELEASED flag will be "set" to
-  indicate the sector is no longer in use.  This is done because the erase
-  block containing the sector cannot necessarily be erased until all sectors
-  in that block have been "released".  This allows sectors in the erase
-  block to remain active while others are inactive until a "garbage collection"
-  operation is needed on the volume to reclaim released sectors.
+  erase state to non-erase state) to indicate that the sector data is valid.
+  When a sector's data needs to be deleted, the RELEASED flag will be "set" to
+  indicate that the sector is no longer in use. This is done because an erase
+  block cannot necessarily be erased until all sectors in that block have been
+  "released". This allows one or more sectors in the erase block to remain active
+  while others are inactive until a "garbage collection" operation is needed on
+  the volume to reclaim released sectors.
 
   The sequence number is used when a logical sector's data needs to be
-  updated with new information.  When this happens, a new physical sector
+  updated with new information. When this happens, a new physical sector
   will be allocated which has a duplicate logical sector number but a
-  higher sequence number.  This allows maintaining flash consistency in the
-  event of a power failure by writing new data prior to releasing the old.
+  higher sequence number. This allows maintaining flash consistency in the
+  event of a sudden power failure by writing new data prior to releasing the old.
   In the event of a power failure causing duplicate logical sector numbers,
-  the sector with the higher sequence number will win, and the older logical
-  sector will be released.
+  the sector with the higher sequence number will be considered as the latest
+  vserion, and given its correctness, the older logical sector will be released.
+
+  The structure of the MTD (Physical) Header is shown below:
+
+	------------------------------------------------------------
```

Comment:
crc 16 is one of option, so you should explain it when you describe regarding journaling